### PR TITLE
.env update

### LINKED ISSUE #638: Prisma CLI and .env file loading (from 2.3.0)
Internal context: https://prisma-company.slack.com/archives/CEYCG2MCN/p1594304960437800

### LINKED ISSUE #639: Clean up .env sections
1. https://www.prisma.io/docs/reference/tools-and-interfaces/prisma-schema#defining-environment-variables-with-the-env-function `env()` is not for "defining environment variables", but to access them in the schema.
3. "If you want environment variables to be evaluated at runtime ..." should be extracted into a new headline "Using your environment variables in your code" as the other part only defines how `.env` files are loaded in the CLI. (This needs clarification if that loads from the same locations as the CLI @Jolg42)
4. https://www.prisma.io/docs/reference/tools-and-interfaces/prisma-schema#use-multiple-env-files-or-env-files-outside-the-prisma-folder is a bit confusing: You _either_ use `dotenv` in your code - OR you install and use `dotenv-cli` - so the full section should be split into "via command line tool" or "via your application code" or similar.

### LINKED ISSUE #659: Local environment variables vs. .env file
From https://www.prisma.io/docs/reference/tools-and-interfaces/prisma-schema#using-environment-variables it is not clear how environment variables and the `.env` file really work - what happens if you have both a file and a value in your local environment? Which one wins and why?

### PR DESCRIPTION
**Pull request by mhwelander**
Closes #638, #639

### CODE REVIEW

**Comment by mhwelander** [comment]
Closes #659

File: content/03-reference/01-tools-and-interfaces/01-prisma-schema/index.mdx [hunk 1]
```diff
@@ -116,22 +118,26 @@ There are a few limitations with `env` at the moment:
 
 ### Using .env files
 
-For many developer tools, it has become a good practice to define environment variables using [`.env`](https://github.com/motdotla/dotenv) files.
+For many developer tools, it has become a good practice to define environment variables using [`.env`](https://github.com/motdotla/dotenv) files. Prisma provides support for `.env` files if:
 
-Prisma provides native support for `.env` files **if the `.env` file is located in the same directory as your Prisma schema file**:
+- The `.env` file is located in the same directory as your Prisma schema file
+- You use the `--schema` option to specify the location of the `.env` file
```
Comment (janpio) [inline content/03-reference/01-tools-and-interfaces/01-prisma-schema/index.mdx]: Both lines here say the same thing: The .env file needs to be next to your schema. By default that is `prisma/schema.prisma`, but can be overridden to whatever you want with `--schema`. In both cases the .env file needs to be next to your schema file.

(The second line is technically not correct - with `--schema` you specify the location of the .env file, which then triggers the CLI to look for the .env file there as well)

Comment (janpio) [reply content/03-reference/01-tools-and-interfaces/01-prisma-schema/index.mdx]: Maybe the right thing to write here is:
1. Next to your schema file
2. Fallback to `prisma/.env` if not found next to schema

File: content/03-reference/01-tools-and-interfaces/01-prisma-schema/index.mdx [hunk 2]
```diff
@@ -116,22 +118,26 @@ There are a few limitations with `env` at the moment:
...
-    ├── .env
-    └── schema.prisma
-```
+The following table describes where Prisma will look for the `.env` file:
+
+| **Command**                                                  | **Schema file location**                                                        | **`.env` file locations checked, in order**    |
+| :----------------------------------------------------------- | :------------------------------------------------------------------------------ | :--------------------------------------------- |
+| `prisma [command] --schema=./folder/subfolder/schema.prisma` | `./folder/subfolder/schema.prisma`                                              | `./folder/subfolder/.env`<br />`./prisma/.env` |
+| `prisma [command]`                                           | `./prisma/schema.prisma`                                                        | `./prisma/.env`                                |
+| `prisma [command]`                                           | No schema (for example, when running `prisma introspect` in an empty directory) | `./prisma/.env`                                |
```
Comment (janpio) [inline content/03-reference/01-tools-and-interfaces/01-prisma-schema/index.mdx]: Maybe list the default case first?
```suggestion
| `prisma [command]`                                           | `./prisma/schema.prisma`                                                        | `./prisma/.env`                                |
| `prisma [command] --schema=./folder/subfolder/schema.prisma` | `./folder/subfolder/schema.prisma`                                              | `./folder/subfolder/.env`<br />`./prisma/.env` |
| `prisma [command]`                                           | No schema (for example, when running `prisma introspect` in an empty directory) | `./prisma/.env`                                |
```

File: content/03-reference/01-tools-and-interfaces/01-prisma-schema/index.mdx [hunk 3]
```diff
@@ -116,22 +118,26 @@ There are a few limitations with `env` at the moment:
...
-This means any environment variables defined in that `.env` file will automatically be loaded when running a Prisma CLI command.
+Any environment variables defined in that `.env` file will automatically be loaded when running a Prisma CLI command.
 
-> **WARNING**: Do not commit your `.env` files into version control.
+> **Warning**: Do not commit your `.env` files into version control.
 
-For example, it is a common scenario to set your database connection URL via an environment variable:
+#### Example: The `DATABASE_URL` environment variable
+
+It is a common to set your database connection URL via an environment variable:
```
Comment (janpio) [inline content/03-reference/01-tools-and-interfaces/01-prisma-schema/index.mdx]:
```suggestion
It is common to load your database connection URL from an environment variable:
```

File: content/03-reference/01-tools-and-interfaces/01-prisma-schema/index.mdx [hunk 4]
```diff
@@ -150,40 +156,53 @@ DATABASE_URL=postgresql://test:test@localhost:5432/test?schema=public
 
 When running any command that needs to access the database defined via the `datasource` block (e.g. `prisma introspect`), the Prisma CLI automatically loads the `DATABASE_URL` environment variables from the `.env` file and makes it available to the CLI.
```
Comment (janpio) [inline content/03-reference/01-tools-and-interfaces/01-prisma-schema/index.mdx]: Line 150 above "This requires ..." is not really correct. The `DATABASE_URL` can also be supplied from the machine itself - the .env file is just another variation. Not sure if this should be mentioned here.

File: content/03-reference/01-tools-and-interfaces/01-prisma-schema/index.mdx [hunk 5]
```diff
@@ -150,40 +156,53 @@ DATABASE_URL=postgresql://test:test@localhost:5432/test?schema=public
...
+dotenv.config() // Load the environment variables
 console.log(`The connection URL is ${process.env.DATABASE_URL}`)
 ```
 
-### Use multiple .env files or .env files outside the prisma folder
+### Manage `.env` files with `dotenv-cli`
 
-To use multiple `.env` files:
+- Use multiple `.env` files
+- Use `.env` files outside the `prisma` folder
```
Comment (janpio) [inline content/03-reference/01-tools-and-interfaces/01-prisma-schema/index.mdx]: The correct wording would probably be "not besides your Prisma schema" or "not in the location where the CLI will look by itself" - see the table above

File: content/03-reference/01-tools-and-interfaces/01-prisma-schema/index.mdx [hunk 6]
```diff
@@ -150,40 +156,53 @@ DATABASE_URL=postgresql://test:test@localhost:5432/test?schema=public
...
-    import { config } from 'dotenv'
-    config({ path: '.env3' })
     ```
+    dotenv -e .env3 npx prisma introspect
+    ```
+
+#### Via application code
 
-    - Use `dotenv` when you run any Prisma command and specify which `.env` file to use. The following example uses a file named `.env3`:
+1.  Install [dotenv-cli](https://www.npmjs.com/package/dotenv-cli):
```
Comment (janpio) [inline content/03-reference/01-tools-and-interfaces/01-prisma-schema/index.mdx]:
```suggestion
1.  Add [dotenv](https://www.npmjs.com/package/dotenv) to your project:
```

File: content/03-reference/01-tools-and-interfaces/01-prisma-schema/index.mdx [hunk 7]
```diff
@@ -150,40 +156,53 @@ DATABASE_URL=postgresql://test:test@localhost:5432/test?schema=public
...
 
     ```
-    dotenv -e .env3 npx prisma introspect
+    npm install -g dotenv-cli
```
Comment (janpio) [inline content/03-reference/01-tools-and-interfaces/01-prisma-schema/index.mdx]:
```suggestion
    npm install dotenv
```

File: content/03-reference/01-tools-and-interfaces/01-prisma-schema/index.mdx [hunk 8]
```diff
@@ -150,40 +156,53 @@ DATABASE_URL=postgresql://test:test@localhost:5432/test?schema=public
...
+    ```
```
Comment (janpio) [inline content/03-reference/01-tools-and-interfaces/01-prisma-schema/index.mdx]: I am not 100% sure this dependency needs to be added to your proejct manually or if it comes already with Prisma Client. @timsuchanek might know.

File: content/03-reference/01-tools-and-interfaces/01-prisma-schema/index.mdx [hunk 9]
```diff
@@ -93,12 +93,14 @@ model User {
...
 ## Using environment variables
 
-You can use environment variables to provide configuration options when a CLI command is invoked. This can e.g. be helpful in order to:
+You can use environment variables to provide configuration options when a CLI command is invoked. Environment variables allow you to:
 
 - Keep secrets out of the schema file
 - Improve portability of the schema file
 
-### Defining environment variables with the env function
+You can use machine-specific variables, `.env` files, or a combination of both. Refer to the `dotenv` documentation for information about [what happens if an environment variable is defined in two places](https://www.npmjs.com/package/dotenv#what-happens-to-environment-variables-that-were-already-set).
```
Comment (janpio) [inline content/03-reference/01-tools-and-interfaces/01-prisma-schema/index.mdx]: I think the "Refer to ..." would better be positioned under "Using .env files" further down, and possibly the explanation really repeated here so people can pick it up from our own docs.

File: content/03-reference/01-tools-and-interfaces/01-prisma-schema/index.mdx (hunk 1 above)
Comment (timsuchanek) [reply content/03-reference/01-tools-and-interfaces/01-prisma-schema/index.mdx]: I agree, the second line is not correct, as you can't provide the .env file with `--schema`.

File: content/03-reference/01-tools-and-interfaces/01-prisma-schema/index.mdx (hunk 8 above)
Comment (mhwelander) [reply content/03-reference/01-tools-and-interfaces/01-prisma-schema/index.mdx]: Not in list of dependencies, so I assume not.

File: content/03-reference/01-tools-and-interfaces/01-prisma-schema/index.mdx [hunk 10]
```diff
@@ -141,49 +149,64 @@ datasource db {
...
 ```ts
 import * as dotenv from 'dotenv'
 
-dotenv.config() // load the environment variables
+dotenv.config() // Load the environment variables
 console.log(`The connection URL is ${process.env.DATABASE_URL}`)
 ```
 
-### Use multiple .env files or .env files outside the prisma folder
+### Manage `.env` files with `dotenv-cli`
```
Comment (janpio) [inline content/03-reference/01-tools-and-interfaces/01-prisma-schema/index.mdx]:
```suggestion
### Manage `.env` files manually
```

File: content/03-reference/01-tools-and-interfaces/01-prisma-schema/index.mdx [hunk 11]
```diff
@@ -141,49 +149,64 @@ datasource db {
...
 
-To use multiple `.env` files:
+You can use `dotenv-cli` if you want to:
```
Comment (janpio) [inline content/03-reference/01-tools-and-interfaces/01-prisma-schema/index.mdx:174]:
```suggestion
You can manage your `.env` files manually if you want to:
```

**Pull request merged**